clang/runtime/dpct-rt/include/dpct/device.hpp

```diff
@@ -717,7 +719,7 @@ class dev_mgr {
         break;
       }
     }
-    _thread2dev_map.clear();
+    _dev_stack = std::stack<unsigned int>();
```

Comment:
Only clear the stack of the current thread, other threads' stacks are unchanged.
Should we clear other stacks together or do nothing?

Reply:
`_dev_stack` is `thread_local` and will take care of this issue.

Reply:
It means only the `_dev_stack` of the current thread can be accessed. `_dev_stack` of other threads are not updated.

Reply:
At this moment we don't have a way to clear all threads. And I can't find a place where this method is used. Shall we do nothing and leave a comment here?

Reply:
Agreed.